code/modules/power/supermatter/supermatter.dm

```diff
@@ -352,45 +352,48 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 /obj/machinery/power/supermatter_crystal/proc/explode()
 	for(var/mob in GLOB.alive_mob_list)
-		var/mob/living/L = mob
-		if(istype(L) && L.z == z)
-			if(ishuman(mob))
-				//Hilariously enough, running into a closet should make you get hit the hardest.
-				var/mob/living/carbon/human/H = mob
-				H.hallucination += max(50, min(300, DETONATION_HALLUCINATION * sqrt(1 / (get_dist(mob, src) + 1)) ) )
-			var/rads = DETONATION_RADS * sqrt( 1 / (get_dist(L, src) + 1) )
-			L.rad_act(rads)
-
-	var/turf/T = get_turf(src)
-	for(var/mob/M in GLOB.player_list)
-		var/turf/mob_turf = get_turf(M)
-		if(T.z == mob_turf.z)
-			SEND_SOUND(M, 'sound/magic/charge.ogg')
-
-			if (M.z == z)
-				to_chat(M, "<span class='boldannounce'>You feel reality distort for a moment...</span>")
-				SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "delam", /datum/mood_event/delam)
-			else
-				to_chat(M, "<span class='boldannounce'>You hold onto \the [M.loc] as hard as you can, as reality distorts around you. You feel safe.</span>")
+		var/mob/living/living = mob
+		if(!istype(living) || living.z != z)
+			continue
+		if(ishuman(mob))
+			//Hilariously enough, running into a closet should make you get hit the hardest.
+			var/mob/living/carbon/human/human = mob
+			human.hallucination += max(50, min(300, DETONATION_HALLUCINATION * sqrt(1 / (get_dist(mob, src) + 1)) ) )
+		var/rads = DETONATION_RADS * sqrt( 1 / (get_dist(living, src) + 1) )
+		living.rad_act(rads)
+
+	var/turf/local_turf = get_turf(src)
+	for(var/mob/mob in GLOB.player_list)
```

Comment:
This should be an `as anything`, but also it still needs the better name as well.